.github/workflows/release.yml

```diff
@@ -1,9 +1,12 @@
 name: Publish Release
 
-on:  
+on:
   push:
     tags:
       - 'v*'
+env:
```

Comment:
Isn't better to keep env clear for the workflow, but provide these env variables for a particular job (aka `yarn build`)?
Then:
- there will be no vulnerability that some GHA can read such ENVs
- it will be obvious who needs these env vars